deploy/awstools/awstools.py

```diff
@@ -220,17 +271,39 @@ def launch_instances(instancetype, count, instancemarket, spotinterruptionbehavi
                 ]),
                 "InstanceMarketOptions":marketconfig,
             }
-            if user_data_file is not None:
+            if user_data_file:
                 with open(user_data_file, "r") as f:
                     instance_args["UserData"] = ''.join(f.readlines())
 
             instance = ec2.create_instances(**instance_args)
             instances += instance
 
         except client.exceptions.ClientError as e:
```

Comment:
We might want to explicitly check for `e.value.response["Error"]["Code"] == "InsufficientInstanceCapacity"` to make it less likely that someone would have a non-capacity `ClientError` and sit around waiting.